### PR TITLE
[Fix #152] Add `AdditionalMethods` configuration to `Minitest/GlobalExpectations`

### LINKED ISSUE #152: Minitest/GlobalExpectations: configure extra matcher methods
## Is your feature request related to a problem? Please describe.

In one of my projects, there are other matcher methods that are not a part of the core Minitest library (perhaps an extension) but which would be nice to include in this cop.

## Describe the solution you'd like

Add a configuration field which includes additional matcher methods for which to check:
```markdown
AdditionalMethods:
  - must_match_array
  - wont_match_array
```

## Describe alternatives you've considered

A custom cop is not maintainable.

## Additional context

No.

### PR DESCRIPTION
**Pull request by tejasbubane**
Closes #152

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop-minitest/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop-minitest/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by tejasbubane** [comment]
@koic can you review this please?

**Comment by koic** [comment]
Can you update the parameter name to `ExtensionMatchers` and mention it in the documentation text?

**Comment by tejasbubane** [comment]
@koic Changed to `ExtensionMatchers`.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
 ### New features
 
 * [#155](https://github.com/rubocop/rubocop-minitest/issues/155): Provide `assert_offense`, `assert_correction`, and `assert_no_offenses` testing APIs for custom Minitest cop development. ([@koic][])
+* [#152](https://github.com/rubocop/rubocop-minitest/issues/152): Add `ExtensionMatchers` configuration to `Minitest/GlobalExpectations`. ([@tejasbubane][])
```
Comment (koic) [inline CHANGELOG.md:10]: Can you use the following new changelog entry format instead?

> Added an entry (file) to the changelog folder named {change_type}_{change_description}.md if the new code introduces user-observable changes. See changelog entry format for details.

https://github.com/rubocop/rubocop-minitest/blob/master/.github/PULL_REQUEST_TEMPLATE.md

Comment (tejasbubane) [reply CHANGELOG.md]: Done

File: docs/modules/ROOT/pages/cops_minitest.adoc [hunk 2]
```diff
@@ -519,7 +519,7 @@ end
 | Yes
 | Yes
 | 0.7
-| 0.16
+| 0.17
```
Comment (koic) [inline docs/modules/ROOT/pages/cops_minitest.adoc:522]: It looks like it contains unnecessary document update.

Comment (tejasbubane) [reply docs/modules/ROOT/pages/cops_minitest.adoc:522]: Shouldn't the `Version Changed` be updated when adding new config option?

Comment (koic) [reply docs/modules/ROOT/pages/cops_minitest.adoc:522]: Yeah, the default behavior with user impact does not change, so `VersionChanged` does not need to be added.

And FYI, this docs/modules/ROOT/pages/cops_minitest.adoc doc is auto-generated at the time of the next release. Please specify `VersionChanged: '<<next>>'` in config/default.yml if the value will be updated. (Not needed in this PR.)

**Comment by tejasbubane** [comment]
@koic ping

**Comment by koic** [comment]
Hm, the approach we should take will depend on what the custom matchers are like. I've asked the issue author @gi the use case. Let's wait for a reply.

**Comment by koic** [comment]
I'm going close this PR because no response for a long time from #152. Flexibility without really necessary extensibility introduces unnecessary complexity at the cost of simplicity.

**Pull request closed without merging**